cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/types/WrapState.kt

```diff
@@ -26,19 +26,39 @@
 package de.fraunhofer.aisec.cpg.graph.types
 
 import de.fraunhofer.aisec.cpg.graph.types.PointerType.PointerOrigin
+import java.util.*
 
-/** Stores State for rewrap when typeinformation has been unwrapped */
+/**
+ * A [WrapState] holds information how a "wrapped" type (for example a [PointerType]) is built from
+ * its element types(s). This can potentially be a chain of different pointer/array operations.
+ */
 class WrapState {
-    @JvmField var depth = 0
+    /** The total depth of "wrapping". This is usually equals to [Type.referenceDepth] */
+    var depth = 0
+
+    /**
+     * An array of [PointerType.PointerOrigin] values, applied in the order the types are wrapped
+     * in.
+     */
+    var pointerOrigins: Array<PointerOrigin?> = arrayOf(PointerOrigin.ARRAY)
```

Comment:
I would probably keep this as-is for now, because I am not sure how much loose in setting up a list instead of an array. We do this wrapping thing quite often.